### PR TITLE
Add a method returns a bool value

### PR DESCRIPTION
**Pull request by K-Sato1995**
# Overview
- Adding `ClamAV:Client#ping` method.
- Adding private `ClamAV:Client#instream` and `ClamAV:Client#scan` methods.
- Adding `ClamAV:Client#safe?` method that returns `true` when all the given files are safe.

# Related Issue
#10

### CODE REVIEW

File: lib/clamav/client.rb [hunk 1]
```diff
@@ -49,5 +49,18 @@ def resolve_default_socket
         ::UNIXSocket.new(unix_socket || '/var/run/clamav/clamd.ctl')
       end
     end
+
+    def ping
+      execute Commands::PingCommand.new
+    end
+
+    def scan(file)
```
Comment (franckverrot) [inline lib/clamav/client.rb]: I believe this should be marked as private as the only method part of the API seems to be `safe?`

File: lib/clamav/client.rb [hunk 2]
```diff
@@ -49,5 +49,18 @@ def resolve_default_socket
...
+      execute Commands::InstreamCommand.new(file) if file.class = StringIO
```
Comment (franckverrot) [inline lib/clamav/client.rb]: `file.class = StringIO` can't work as `#class=` isn't a method on any known class in Ruby's API.  I think that what you want to test is equality between that file's type and `StringIO`.  This could be done using `is_a?`.

Also, chaining calls to `execute` won't make `scan` return the right value (if `file` is indeed a StringIO, you'll still end up executing the line below, and the method will return whatever the line below returns.

You could want to use `if ... else` here.

File: lib/clamav/client.rb [hunk 3]
```diff
@@ -49,5 +49,18 @@ def resolve_default_socket
...
+      execute Commands::ScanCommand.new(file)
+    end
+
+    def safe?(file)
```
Comment (franckverrot) [inline lib/clamav/client.rb]: `scan` is likely to return an array right now, so calling `#virus_name` here is likely to raise an exception.

**Review by franckverrot — CHANGES_REQUESTED** [review]
Thanks for the PR!

There are a few blocking issues with the code.  How do you think about starting with `ping`, and move onto the method you really wanted? (`safe?`) ?  If this works for you, let's add tests right away in the `test` folder.

In order to run the tests without having a working Ruby + clamav setup in your development environment, I just added support for Docker (to get CI to pass).  How to use it?

Let's build the docker image:
`RUBY_VERSION=2.6 docker build -f test/Dockerfile --build-arg RUBY_VERSION -t my-clamav-client .`

Now let's run the tests.  From wherever you clone this project, you can run:

`docker run -it --rm -v $PWD:/clamav-client my-clamav-client rake`

In short (and with tons of simplifications):
* `-it` will enable you to type in the container if you replaced `rake` with `bash`.  It isn't super useful to run `rake` but let's keep it
* `--rm` will remove the data associated with that container upon exit
* `-v ...` will bind mount your local directory within the container, super useful if you were running `bash` instead of `rake` and then run `rake` from within the container, no need to restart it all the time!

Tell me if you want to continue on this PR.  I can also work on it at some point.

Thanks!
Franck

**Comment by K-Sato1995** [comment]
@franckverrot 
Thanks for reviewing the code!

> Tell me if you want to continue on this PR. I can also work on it at some point.

I should have some time to actually work on this sometime soon!!

**Comment by K-Sato1995** [comment]
@franckverrot 
Hi! I updated my PR.
I'd appreciate if you review it again🙇🏼‍♀️. 
Thanks!

**Comment by timoschilling** [comment]
@franckverrot can we get this merged now?

**Comment by franckverrot** [comment]
Just released 3.2.0.  Thanks @K-Sato1995
